src/Data/Vec/Functional/Algebra/Properties.agda

```diff
@@ -0,0 +1,725 @@
+    { isRightSemimodule = isRightSemimodule
+    ; -ᴹ‿cong           = -ᴹ‿cong -‿cong
+    ; -ᴹ‿inverse        = -ᴹ‿inverse -‿inverse
+    }
+
+  isBimodule : IsBimodule ring ring RNS._≈_ RNS._+_ RNS.0# VG._⁻¹ LSM._*ₗ_ RSM._*ᵣ_
+  isBimodule = record
+    { isBisemimodule = isBisemimodule
+    ; -ᴹ‿cong        = -ᴹ‿cong -‿cong
+    ; -ᴹ‿inverse     = -ᴹ‿inverse -‿inverse
+    }
+
+  isLeftModule : IsLeftModule ring RNS._≈_ RNS._+_ RNS.0# VG._⁻¹ LSM._*ₗ_
+  isLeftModule = record
+    { isLeftSemimodule = isLeftSemimodule
+    ; -ᴹ‿cong          = -ᴹ‿cong -‿cong
+    ; -ᴹ‿inverse       = -ᴹ‿inverse -‿inverse
+    }
+
+  +ᴹ-*-isRing : IsRing RS._≈_ RS._+_ RS._*_ VG._⁻¹ RS.0# RS.1#
+  +ᴹ-*-isRing = record
+    { +-isAbelianGroup = +ᴹ-isAbelianGroup
+    ; *-cong           = λ x≈y u≈v i → *-cong (x≈y i) (u≈v i)
+    ; *-assoc          = λ xs ys zs i → *-assoc (xs i) (ys i) (zs i)
+    ; *-identity       = let idₗ , idᵣ = *-identity in
+        (λ xs i → idₗ (xs i)) , (λ xs i → idᵣ (xs i))
+    ; distrib          = let dₗ , dᵣ = distrib in
+        (λ xs ys zs i → dₗ (xs i) (ys i) (zs i))
+      , (λ xs ys zs i → dᵣ (xs i) (ys i) (zs i))
+    }
+
+  leftModule : LeftModule _ _ _
+  leftModule = record { isLeftModule = isLeftModule }
+
+  bisemimodule : Bisemimodule _ _ _ _
+  bisemimodule = record { isBisemimodule = isBisemimodule }
+
+  rightModule : RightModule _ _ _
+  rightModule = record { isRightModule = isRightModule }
+
+  bimodule : Bimodule _ _ _ _
+  bimodule = record { isBimodule = isBimodule }
+
+------------------------------------------------------------------------
+
+module CommutativeRingProperties (commutativeRing : CommutativeRing a ℓ) (n : ℕ) where
+  open CommutativeRing commutativeRing
+  private
+    module Ri = Ring ring
+    module S  = Semiring Ri.semiring
+    module SR = RawSemiring S.rawSemiring
+    baseNearSemiring = SR.rawNearSemiring
+
+    nearSemiringᵛ = rawNearSemiringᵛ baseNearSemiring n
+    semiringᵛ     = rawSemiringᵛ     S.rawSemiring    n
+
+    module RNS = RawNearSemiring nearSemiringᵛ
+    module RS  = RawSemiring     semiringᵛ
+
+    leftSemiᵛ  = rawLeftSemimoduleᵛ  baseNearSemiring n
+    rightSemiᵛ = rawRightSemimoduleᵛ baseNearSemiring n
+    module LSM = RawLeftSemimodule  leftSemiᵛ
+    module RSM = RawRightSemimodule rightSemiᵛ
+
+    module +G = Group Ri.+-group
+    addRawGroup = +G.rawGroup
+    addGroupᵛ = rawGroupᵛ addRawGroup n
+    module VG   = RawGroup addGroupᵛ
+
+  open RingProperties ring n public
+
+  +ᴹ-*-isCommutativeRing
+    : IsCommutativeRing RS._≈_ RS._+_ RS._*_ VG._⁻¹ RS.0# RS.1#
+  +ᴹ-*-isCommutativeRing = record
+    { isRing = +ᴹ-*-isRing
+    ; *-comm = λ xs ys i → *-comm (xs i) (ys i)
+    }
+
+  isModule : IsModule commutativeRing RNS._≈_ RNS._+_ RNS.0# VG._⁻¹ LSM._*ₗ_ RSM._*ᵣ_
+  isModule = record
+    { isBimodule = isBimodule
+    ; *ₗ-*ᵣ-coincident = λ r xs i → *-comm r (xs i)
+    }
+
+  module' : Module _ _ _
+  module' = record { isModule = isModule }
```

Comment:
For the underlying name of the bundle, `stdlib` uses in eg. `Algebra.Module.Construct.TensorUnit` the name `⟨module⟩`, and I would follow suit here, for the sake of consistency.